core/asset/block.go

```diff
@@ -32,11 +32,16 @@ func (reg *Registry) indexAnnotatedAsset(ctx context.Context, a *Asset) error {
 	m := map[string]interface{}{
 		"id":               a.AssetID,
 		"alias":            a.Alias,
-		"definition":       a.Definition,
+		"raw_definition":   json.HexBytes(a.RawDefinition()),
+		"vm_version":       a.VMVersion,
 		"issuance_program": json.HexBytes(a.IssuanceProgram),
 		"tags":             a.Tags,
 		"is_local":         "no",
 	}
+	adef, err := a.Definition()
+	if err == nil {
+		m["definition"] = adef
```

Comment:
Sorry for the late response here. @oleganza's description fits my understanding--if the definition is not valid JSON, the create-asset API call will fail outright.

Also, following Oleg, I think the sensible default is `nil`. It doesn't really differentiate from the case where the asset definition is empty, but it's better than the alternatives I can think of.

This conversation does suggest that asset defs as exposed by the SDK may be a bit too high-level; previous incarnations of the API exposed byte arrays and left the client to perform JSON serialization/deserialization. On the other hand, tx annotation based on the JSON value was probably what drove the change. Exposing `raw_definition` seems like a good compromise.